front/app/containers/Admin/pagesAndMenu/EditHomepage/HeroBanner/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+
+// components
+import GenericHeroBannerForm, {
+  HeroBannerInputSettings,
+} from '../../containers/GenericHeroBannerForm';
+import {
+  pagesAndMenuBreadcrumb,
+  homeBreadcrumb,
+} from 'containers/Admin/pagesAndMenu/breadcrumbs';
+
+// resources
+import useHomepageSettings from 'hooks/useHomepageSettings';
+import {
+  IHomepageSettingsAttributes,
+  updateHomepageSettings,
+} from 'services/homepageSettings';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import { forOwn, isEqual } from 'lodash-es';
+
+// i18n
+// change
+import messages from '../../containers/GenericHeroBannerForm/messages';
+import { InjectedIntlProps } from 'react-intl';
+import { injectIntl } from 'utils/cl-intl';
+import { ISubmitState } from 'components/admin/SubmitWrapper';
+
+const EditHomepageHeroBannerForm = ({
+  intl: { formatMessage },
+}: InjectedIntlProps) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [formStatus, setFormStatus] = useState<ISubmitState>('disabled');
+
+  const homepageSettings = useHomepageSettings();
+
+  if (isNilOrError(homepageSettings)) {
+    return null;
+  }
+
+  const { attributes } = homepageSettings;
+
+  const handleSave = async (newSettings: HeroBannerInputSettings) => {
+    if (!newSettings) return;
+
+    // some names in the generic form are different from the property on Homepage
```

Comment:
The names of properties are different between HomePage and CustomPage (since CustomPage has no signed in/out concept, it's simpler) so we have to map them to the generic form somehow. Not sure this is the best way but it seems to work